00-Login-Kt/app/src/main/java/com/auth0/sample/MainActivity.kt

```diff
@@ -96,11 +136,71 @@ class MainActivity : AppCompatActivity() {
                     }
 
                     override fun onSuccess(profile: UserProfile?) {
-                        binding.userProfile.setText(
-                                "Name: ${profile?.name}\n" +
-                                "Email: ${profile?.email}")
+                        cachedUserProfile = profile;
+                        updateUI()
                     }
         }) }
     }
 
+    private fun getUserMetadata() {
+        // Get the access token so that we can make calls to the management API
+        cachedCredentials?.accessToken?.let { accessToken ->
+            // Get the user ID and call the full getUser Management API endpoint, to retrieve the full profile information
+            cachedUserProfile?.getId()?.let { userId ->
+                // Create the user API client
+                val usersClient = UsersAPIClient(account, accessToken)
+
+                // Get the full user profile
+                usersClient.getProfile(userId).start(object: Callback<UserProfile, ManagementException> {
+                    override fun onFailure(exception: ManagementException) {
+                        Snackbar.make(
+                                binding.root,
+                                "Failure: ${exception.getCode()}",
+                                Snackbar.LENGTH_LONG
+                        ).show()
+                    }
+
+                    override fun onSuccess(userProfile: UserProfile?) {
+                        cachedUserProfile = userProfile;
+                        updateUI()
+                        binding.inputEditMetadata.setText(cachedUserProfile?.getUserMetadata()?.get("country") as String? ?: "")
```

Comment:
@lbalmaceda Can you confirm whether or not there's a better way to write this?

Specifically the `cachedUserProfile?.getUserMetadata()?.get("country") as String? ?: ""` part.

Reply:
If the variable is not-null you don't need to access it safely with the `?` (I see you do that after `getUserMetadata`).

I'd probably move it to a separate line to make it easier to follow. Also, the TextView supports being set a `null` text and would use "" internally like you did above. 

![image](https://user-images.githubusercontent.com/3900123/104731839-9686c100-573c-11eb-96e5-da3a6b49fd9d.png)

I'd change it to this. You can assume the profile is not null.
```kotlin
val country = cachedUserProfile!!.getUserMetadata().get("country")
```

Reply:
Good point. The reason I've used `getUserMetadata()?.get` is because otherwise the IDE complains about it being an unsafe call:

![image](https://user-images.githubusercontent.com/766403/104744630-013bfa80-5745-11eb-947a-5e3d7b824e0b.png)

Despite the return type of `getUserMetadata` being `Map<String, Any>`. Do you see the same on your environment?

Reply:
I also can't do this:

```kotlin
val country: String? = cachedUserProfile?.getUserMetadata()?.get("country")
binding.inputEditMetadata.setText(country)
```

Or this:

```kotlin
val country = cachedUserProfile?.getUserMetadata()?.get("country")
binding.inputEditMetadata.setText(country)
```

In both cases it (understandably) can't implicitly cast from `Any?` to `String?`

Reply:
That's odd.. I see the getter returns a non-null value https://github.com/auth0/Auth0.Android/blob/v2-dev/auth0/src/main/java/com/auth0/android/result/UserProfile.kt#L67-L69. Please check that the library version you are using is on the latest changes.

About the cast, you were right. You **need** to cast it because the collection type is `Any`.

This works for me (property accessor). Keeping the `get("country")` works as well.

```kotlin
val c = profile.getUserMetadata()["country"] as String
```

Reply:
I've used this code in the end:

```kotlin
val country = profile!!.getUserMetadata()["country"] as String? ?: ""
```

The elvis operator is still needed in case you try to get the `country` metadata when it doesn't exist.